internal/datatypes/core_test.go

```diff
@@ -7,54 +7,40 @@ import (
-	{"bad data", int64(1), true, enumValidatorFunc},
-	{0.6, int64(0), false, enumValidatorFunc},
-	{6, int64(6), false, enumValidatorFunc},
-	{"test", "test", false, stringValidatorFunc},
-	{5, "test", true, stringValidatorFunc},
-	{"5", "5", false, stringValidatorFunc},
-	{"test", "test", false, textValidatorFunc},
-	{5, "test", true, textValidatorFunc},
-	{"5", "5", false, textValidatorFunc},
-	{"andrew.wansley@gmail.com", "andrew.wansley@gmail.com", false, emailAddressValidatorFunc},
-	{"wansley", "andrew.wansley@gmail.com", true, emailAddressValidatorFunc},
-	{"wansley@", "andrew.wansley@gmail.com", true, emailAddressValidatorFunc},
-	{"wansley@.c", "andrew.wansley@gmail.com", true, emailAddressValidatorFunc},
-	{"chase@a.c", "chase@a.c", false, emailAddressValidatorFunc},
-	{"d9c59e23-e050-4fc7-949d-8535ae8e3a49",getFirst(uuid.Parse("d9c59e23-e050-4fc7-949d-8535ae8e3a49")),false, uuidValidatorFunc},
-	{"d9c59e23-e050-4fc7-949d-8535ae8e3a4","d9c59e23-e050-4fc7-949d-8535ae8e3a49",true,uuidValidatorFunc},
-	{321321,"d9c59e23-e050-4fc7-949d-8535ae8e3a49",true,uuidValidatorFunc},
-	{getFirst(uuid.Parse("d9c59e23-e050-4fc7-949d-8535ae8e3a49")),getFirst(uuid.Parse("d9c59e23-e050-4fc7-949d-8535ae8e3a49")),false, uuidValidatorFunc},
-	{1, float64(1), false, floatValidatorFunc},
-	{"bad data", float64(1), true, floatValidatorFunc},
-	{0.6, 0.6, false, floatValidatorFunc},
-	{int(6), float64(6), false, floatValidatorFunc},
-	{"https://www.google.com","https://www.google.com", false, URLValidatorFunc},
-	{"https://www.google.comdaddy","https://www.google.comdaddy", false, URLValidatorFunc},
-	{"www.google.com","www.google.com", true, URLValidatorFunc},
-	{"localhost:8080","www.google.com", true, URLValidatorFunc},
-	{5,"www.google.com", true, URLValidatorFunc},
-}
-
-func getFirst(a, b interface{}) interface{}{
-	return a
+	{false, false, false, BoolFromJSON},
+	{"bad data", false, true, BoolFromJSON},
+	{0, false, true, BoolFromJSON},
+	{1, int64(1), false, IntFromJSON},
+	{"bad data", int64(1), true, IntFromJSON},
+	{0.6, int64(0), false, IntFromJSON},
+	{6, int64(6), false, IntFromJSON},
+	{1, int64(1), false, EnumFromJSON},
+	{"bad data", int64(1), true, EnumFromJSON},
+	{0.6, int64(0), false, EnumFromJSON},
+	{6, int64(6), false, EnumFromJSON},
+	{"test", "test", false, StringFromJSON},
+	{5, "test", true, StringFromJSON},
+	{"5", "5", false, StringFromJSON},
+	{"test", "test", false, TextFromJSON},
+	{5, "test", true, TextFromJSON},
+	{"5", "5", false, TextFromJSON},
+	{"d9c59e23-e050-4fc7-949d-8535ae8e3a49", getFirst(uuid.Parse("d9c59e23-e050-4fc7-949d-8535ae8e3a49")), false, UUIDFromJSON},
+	{"d9c59e23-e050-4fc7-949d-8535ae8e3a4", "d9c59e23-e050-4fc7-949d-8535ae8e3a49", true, UUIDFromJSON},
+	{321321, "d9c59e23-e050-4fc7-949d-8535ae8e3a49", true, UUIDFromJSON},
+	{getFirst(uuid.Parse("d9c59e23-e050-4fc7-949d-8535ae8e3a49")), getFirst(uuid.Parse("d9c59e23-e050-4fc7-949d-8535ae8e3a49")), false, UUIDFromJSON},
```

Comment:
I think you just want uuid.MustParse, which only returns the uuid and panics on failure: https://godoc.org/github.com/google/uuid#MustParse